core/commonMain/src/kotlinx/io/Input.kt

```diff
@@ -340,93 +371,178 @@ public abstract class Input : Closeable {
         }
 
         // Nope, doesn't fit in a buffer, read byte by byte
-        var long = 0L
+        var result = 0L
         readBytes(primitiveSize) {
-            long = (long shl 8) or it.toLong()
+            result = (result shl 8) or it.toLong()
         }
-        return fromLong(long)
+
+        return result
     }
 
     /**
      * Reads [size] unsigned bytes from an Input and calls [consumer] on each of them.
      * @throws EOFException if no more bytes available.
      */
-    private inline fun readBytes(size: Int, consumer: (unsignedByte: Int) -> Unit) {
+    private fun readBytes(size: Int, consumer: (unsignedByte: Int) -> Unit) {
```

Comment:
This method is no longer inline, but still accepts the lambda that manipulates with primitive.
It has the only caller, so please just `comsumer` logic from a call site into this method

Reply:
I'll revert the `inline` modifier for methods with `block`. It was for measure purpose only